aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -0,0 +1,106 @@
+#include "ATen/native/cpu/TensorCompareKernel.h"
+
+#include <numeric>
+#include <iterator>
+#include <algorithm>
+
+#include "ATen/Dispatch.h"
+#include "ATen/Parallel.h"
+#include "ATen/optional.h"
+
+namespace at { namespace native { namespace {
+
+template <typename scalar_t>
+bool _isnan(scalar_t val) {
+  return false;
+}
+
+template <>
+bool _isnan(float val) {
+  return std::isnan(val);
```

Comment:
Can you just use `std::nan()` directly? It works on integer types and GCC and clang optimize it to `false`/`0` on integers at `-O1` or higher.

Reply:
sorry, i don't quite follow here. Could you please be more explicit "how to use `std::nan()`" to tell whether `val` is NaN or not?

this logic is in order to pass [test_max()](https://github.com/pytorch/pytorch/blob/master/test/test_torch.py#L697) from `test_torch.py`, i was following [THTensor_(max)](https://github.com/pytorch/pytorch/blob/master/aten/src/TH/generic/THTensorMoreMath.cpp#L74) basically.